examples/lib/stories/effects/dual_effect_removal_example.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:flame/components.dart';
+import 'package:flame/effects.dart';
+import 'package:flame/game.dart';
+import 'package:flame/input.dart';
+import 'package:flutter/material.dart';
+import '../../commons/ember.dart';
+
+class DualEffectRemovalExample extends FlameGame with TapDetector {
+  static const String description = '''
+    In this example we show how a dual effect can be used.
+    And removed.To remove just tap anywhere on the screen.
+    1st tap will remove Opacity Effect 2nd tap removes the color effect.    
+  ''';
+
+  late final SpriteComponent mySprite;
+  late  ColorEffect fx;
+  late OpacityEffect opfx;
+  int count = 0;
+  EffectController controller1 = EffectController(duration: 2,reverseDuration: 2,infinite: true,);
+  EffectController controller2 = EffectController(duration: 1,reverseDuration: 1,infinite: true,);
+
+  @override
+  Future<void> onLoad() async {
+    await super.onLoad();
+
+    Sprite flameSprite = await loadSprite('flame.png');
+    Vector2 sz = flameSprite.srcSize;
+    Vector2 pos = Vector2(50,50);
+    mySprite = SpriteComponent(sprite: flameSprite,position: pos,size: sz);
+    add(mySprite);
+    fx = ColorEffect(Colors.blue, const Offset(0.0, 0.8,), controller1 );
+    mySprite.add(fx);
+    opfx = OpacityEffect.fadeOut(controller2);
+    mySprite.add(opfx);
+
+
+  }
+
+  @override
+  void onTap() {
+    count++;
+    if (count > 1) {
+      fx.apply(0);

```

Comment:
Why is `apply` needed here?

Reply:
See https://github.com/flame-engine/flame/issues/1287#issuecomment-1003219515

Reply:
Couldn't this be fixed by having the effects doing this in onRemove? Feels strange that the user should have to do this.

Reply:
Well, "normal" behavior is for an effect to modify its target, so that the target stays modified. Here, however, we want to remove the effect and restore the target to its original state.

Reply:
Aah, now I get it. Feels like there should be a convenience function for this at least, it's probably quite a common usecase.